src/pg_shard.c

```diff
@@ -2104,50 +2111,8 @@ PgShardProcessUtility(Node *parsetree, const char *queryString,
 	}
 	else if (statementType == T_CopyStmt)
 	{
-		CopyStmt *copyStatement = (CopyStmt *) parsetree;
-		RangeVar *relation = copyStatement->relation;
-		Node *rawQuery = copyObject(copyStatement->query);
-
-		if (relation != NULL)
-		{
-			bool failOK = true;
-			Oid tableId = RangeVarGetRelid(relation, NoLock, failOK);
-			bool isDistributedTable = false;
-
-			Assert(rawQuery == NULL);
-
-			isDistributedTable = IsDistributedTable(tableId);
-			if (isDistributedTable)
-			{
-				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
But I still have top copy most of this code to PgShardCopy, because in any case I need "relation", "rawQuery", "tableId" which are extracted here.
Sorry, but I still think that placing all this logic in one place is more important then minimizing changes of existed pg_shard.c code or reducing level of nesting.